buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -375,6 +393,22 @@ public void extraConfigFile(String destination, File from) {
         extraConfigFiles.put(destination, from);
     }
 
+    @Override
+    public void user(Map<String, String> userSpec) {
```

Comment:
I wasn't sure we should validate to this extent or rely on elasticsearch validating the config and cluster not starting if miss-configured in this case. 
I wasn't sure it's worth maintaining checks in testclsuters if all they buy us is that it will fail a bit faster when the config is not right ( which is the exception not the rule ). 
On the other hand, one could use the integ test distro with the right plugins and modules and make security work I suppose. We don't really do this for tests in the x-pack folder, but I think as we move more towards build cache and incremental testing we will need to think of ways to make the tests more modular and dependent only on the parts they are actually testing. Right now with the default distro a test always needs to be ran, but  integ test + a few modules might lead to better cache hits. 

I'm fine either way, don't mind changing if you think we must have this validation in place.